spring-cloud-gcp-starters/spring-cloud-gcp-starter-sql/src/main/java/org/springframework/cloud/gcp/sql/autoconfig/GcpCloudSqlAutoConfiguration.java

```diff
@@ -75,46 +76,35 @@ protected SQLAdmin sqlAdminService(CredentialsProvider credentialsProvider)
 
 		return new SQLAdmin.Builder(httpTransport, jsonFactory,
 				new HttpCredentialsAdapter(credentialsProvider.getCredentials()))
-				.build();
+						.build();
 	}
 
 	@Bean
-	@ConditionalOnMissingBean(CloudSqlJdbcUrlProvider.class)
-	public CloudSqlJdbcUrlProvider cloudSqlJdbcUrlProvider(SQLAdmin sqlAdmin,
-			GcpProjectIdProvider projectIdProvider)
-			throws IOException, ClassNotFoundException {
-		Assert.hasText(projectIdProvider.getProjectId(),
-				"A project ID must be provided.");
-
-		// TODO(joaomartins): Set the credential factory to be used by SocketFactory when
-		// https://github.com/GoogleCloudPlatform/cloud-sql-jdbc-socket-factory/issues/41 is
-		// resolved. For now, it uses application default creds.
-
-		// Load MySQL driver to register it, so the user doesn't have to.
-		Class.forName("com.mysql.jdbc.Driver");
-
-		// Look for the region if the user didn't specify one.
-		String region = this.gcpCloudSqlProperties.getRegion();
-		if (!StringUtils.hasText(region)) {
-			region = sqlAdmin.instances().get(projectIdProvider.getProjectId(),
-					this.gcpCloudSqlProperties.getInstanceName()).execute().getRegion();
-		}
-
-		String instanceConnectionName = projectIdProvider.getProjectId() + ":" + region + ":"
-				+ this.gcpCloudSqlProperties.getInstanceName();
+	@ConditionalOnMissingBean(CloudSqlJdbcInfoProvider.class)
+	@Conditional(AppEngineCondition.class)
```

Comment:
That's easy to check:
````java
@SpringBootConfiguration
public class OverrideApplication {

	public static void main(String[] args) {
		ConfigurableApplicationContext applicationContext = new SpringApplicationBuilder(OverrideApplication.class)
				.run(args);
		Foo bean = applicationContext.getBean(Foo.class);
		System.out.println(bean.getFoo());

		applicationContext = new SpringApplicationBuilder(OverrideApplication.class)
				.properties("bar=bar")
				.run(args);
		bean = applicationContext.getBean(Foo.class);
		System.out.println(bean.getFoo());
	}

	@Bean
	@ConditionalOnMissingBean
	@ConditionalOnProperty("bar")
	public Foo fooConditional() {
		return new Foo("bar");
	}


	@Bean
	@ConditionalOnMissingBean
	public Foo fooDefault() {
		return new Foo("foo");
	}

	public static final class Foo {

		private final String foo;

		public Foo(String foo) {
			this.foo = foo;
		}

		public String getFoo() {
			return this.foo;
		}

	}

}
````

The first one prints `foo` and second `bar` - as expected.
Therefore your solution is correct.